torch/csrc/jit/passes/symbolic_shape_analysis.cpp

```diff
@@ -43,6 +43,91 @@ static bool symbolic_shape_analysis_test_mode = false;
 namespace torch {
 namespace jit {
 
+// This is similar to c10::SymbolicShape, but instead of either having
+// a concrete dimension or a symbolic dimension, an argument may be:
+// - A Symbolic Dimension
+// - A Constant Integer
+// - Neither of the above. The third case can occur due to inputs to
+// ops like view that accept negative values. Maintaining the distinction
+// between an unknown symbolic dimension and an unknown integer allows
+// us to optimize out comparisons to values < 0 (symbolic shapes are always >=
+// 0) For example, a call like graph(%y: Tensor(SS(-1), 10, 10), %inp: int):
```

Comment:
where does a comparison happen in this example? I'm a little bit unclear how symbolic shapes are always >= 0 when they are backed up by negative integers?

Reply:
symbolic shapes represent dimension values. You cannot create a dimension < 0, it would be an error. That they are stored as negative integers is an implementation detail as you said.

Reply:
got it! yes, a symbolic dimension should still be a positive number since it's supposed to stand for some concrete dimension value.